File: backend/PlugPlay/PlugPlay.Api/Dto/Product/FilterProductsResponse.cs
namespace PlugPlay.Api.Dto.Product;

public class FilterProductsResponse
{
    public IEnumerable<ProductDto> Products { get; set; }

    public int Total { get; set; }

    public int TotalPages { get; set; }

    public int Page { get; set; }

    public int PageSize { get; set; }
}

File: mobile/app/src/main/java/com/plugplay/plugplaymobile/data/model/FilterProductsResponse.kt
package com.plugplay.plugplaymobile.data.model

import com.google.gson.annotations.SerializedName

/**
 * DTO для відповіді на ендпоінт фільтрації /api/Products/filter/{categoryId}.
 * Відповідає структурі C# DTO PlugPlay.Api.Dto.Product.FilterProductsResponse.
 */
data class FilterProductsResponse(
    @SerializedName("products")
    val products: List<ProductDto>,

    @SerializedName("total")
    val total: Int,

    @SerializedName("totalPages")
    val totalPages: Int,

    @SerializedName("page")
    val page: Int,

    @SerializedName("pageSize")
    val pageSize: Int
)